### PR TITLE
Update to voxpupuli-test 5

### PR DESCRIPTION
**Pull request by ekohl**
I'm still a bit unsure about the fact handling. https://puppet.com/docs/puppet/7/lang_facts_builtin_variables.html is a very useful resource, but I'm not entirely sure what the intended values should be and how to deal with it in various environments. See the TODOs inline.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -33,19 +33,15 @@
...
+  $srv_domain = fact('networking.domain')
 
   # lint:ignore:puppet_url_without_modules
   $pluginsource        = 'puppet:///plugins'
   $pluginfactsource    = 'puppet:///pluginfacts'
   # lint:endignore
   $classfile           = '$statedir/classes.txt'
   $syslogfacility      = undef
-  $environment         = $::environment
+  $environment         = $server_facts['environment']
```
Comment (ekohl) [inline manifests/params.pp:44]: While https://www.puppet.com/docs/puppet/7/lang_facts_builtin_variables.html#lang_facts_builtin_variables-server-facts doesn't mention it, `$server_facts['environment']` is present when using `puppet apply`.

File: manifests/params.pp [hunk 2]
```diff
@@ -198,13 +194,8 @@
...
   # Will this host be a puppet agent ?
   $agent                      = true
-  $client_certname            = $::clientcert
-
-  if defined('$::puppetmaster') {
-    $puppetmaster             = $::puppetmaster
-  } else {
-    $puppetmaster             = undef
-  }
+  $client_certname            = $trusted['certname']
```
Comment (ekohl) [inline manifests/params.pp:197]: This is also present when using `puppet apply`

File: manifests/params.pp [hunk 3]
```diff
@@ -198,13 +194,8 @@
...
+  $puppetmaster               = $server_facts['servername']
```
Comment (ekohl) [inline manifests/params.pp:198]: This is one I wonder about. `puppetmaster` is a global variable we set in Foreman's ENC, which allows you to change the puppetserver easily. Also, setting this outside of Foreman's ENC. For example, when using DNS lookups. Should we keep the old `defined` code/update it to `getvar()`?

**Review by ekohl — COMMENTED** [review]
@alexjfisher @bastelfreak would you mind taking a look at these comments?

File: manifests/params.pp (hunk 3 above)
Comment (ekohl) [reply manifests/params.pp]: I ended up changing it to `getvar('puppetmaster')`, to remain compatible.

File: spec/support/trusted.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+RSpec.configure do |c|
+  c.trusted_node_data = true
+  c.trusted_server_facts = true
```
Comment (ekohl) [inline spec/support/trusted.rb:3]: Surprisingly https://github.com/puppetlabs/rspec-puppet#trusted_node_data and https://github.com/puppetlabs/rspec-puppet#trusted_server_facts are off by default.

File: spec/classes/puppet_server_spec.rb [hunk 5]
```diff
@@ -503,6 +496,7 @@
           end
 
           it 'should not sync the crl' do
+            pending("rspec-puppet always sets $server_facts['servername']")
```
Comment (ekohl) [inline spec/classes/puppet_server_spec.rb:499]: https://github.com/puppetlabs/rspec-puppet/issues/37

**Review by tuxmea — APPROVED** [review]

**Comment by ekohl** [comment]
No content change compared to the previous version, just extracted part of it to https://github.com/theforeman/puppet-puppet/pull/841 and rebased it on top. Also added a link to the rspec-puppet GH issue in the code itself.

**Pull request merged**